### DISCUSSION TITLE
Data render based on boolean data intial state true

### DISCUSSION THREAD

**Discussion by algospa** [TanStack/table#5308] [Q&A]
> Version 8.11.3

After i made a custom filter toggle to show and switch between true and false for showing the available status of my item in my table (wich is set true, but only on the checked property of the switch), my table as expected is showing all items at first with status true and false, how can i make possible to set the intialstate to be getting the column available (the sattus) from my table and set the intialState to be true, so it can show only the available items?

Do i need to change the layout of the table or do i need to edit my filter of filterFn? im new to tanstack table

**Comment by algospa** [comment]
i found a sort of solution and thats needs to be applying the intialfilter to eb set to the value you want and the id of the column

```
initialState: {
  pagination: {
    pageSize: 20,
  },
  columnFilters: [
    {
      id: 'available',
      value: true
    },
  ],
},
```

2 problems occured with this:
- if i set it into initialState object, it will not work
- if i set it to the state object, it will reduce to an infite loop, crashing my app instantly (please fix at this point)

**Reply by talonx** [reply]
>  if i set it into initialState object, it will not work
>  if i set it to the state object, it will reduce to an infite loop, crashing my app instantly (please fix at this point)

I ran into these same 2 problems. Did you figure out how to resolve them? The "officially" correct way seems to be to set the columnFilters in the initialState but that does not seem to work.
